.github/workflows/ubuntu-ci-c6a-x86_64.yaml

```diff
@@ -0,0 +1,210 @@
+on:
+  pull_request:
+    paths-ignore:
+      - 'doc/**'
+      - '**.md'
+      - '.github/ISSUE_TEMPLATE/*'
+      - '.gitignore'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  ubuntu-ci-c6a-x86_64-build:
+    runs-on: [ubuntu-ci-c6a-x86_64]
+
+    steps:
+      - name: cleanup
+        run: |
+          pwd
+          ls -lart
+          rm -fr *
+
+      - name: checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      - name: create-env
+        run: |
+          source ./setup.sh
+
+          # DH* 20230302 - to avoid using padded_length for build caches,
+          # always build in the same environment so that the length of the
+          # path doesn't change - see also other 'DH* 20230302' changes below
+          #export ENVNAME=${{ inputs.template || 'unified-dev' }}.ubuntu-ci-x86_64
+          export ENVNAME=ci-env.ubuntu-ci-c6a-x86_64
+          # *DH
+          export ENVDIR=$PWD/envs/${ENVNAME}
+          spack stack create env --site linux.default --template ${{ inputs.template || 'unified-dev' }} --name ${ENVNAME}
+          spack env activate ${ENVDIR}
+          spack add ${{ inputs.specs || '' }}
+          export SPACK_SYSTEM_CONFIG_PATH="${ENVDIR}/site"
+
+          # Find external packages
+          spack external find --scope system \
+              --exclude bison --exclude openssl \
+              --exclude curl --exclude python
+          spack external find --scope system sed
+          spack external find --scope system perl
+          spack external find --scope system wget
+          PATH="/usr/local/opt/qt5/bin:$PATH" \
+              spack external find --scope system qt
+          spack external find --scope system texlive
+          spack external find --scope system mysql
+
+          # For GNU
+          # TODO - anything needed?
+
+          # For Intel
+          # Need to find external Intel MPI and annotate with the
+          # correct compiler, no way to do that with spack commands.
+          # echo "" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
+          # echo "  intel-oneapi-mpi:" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
+          # echo "    buildable: false" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
+          # echo "    externals:" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
+          # echo "    - spec: intel-oneapi-mpi@2021.6.0%intel@2021.6.0" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
+          # echo "      prefix: /opt/intel" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
+          # echo "      modules:" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
+          # echo "      - libfabric-aws/1.16.0~amzn4.0" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
+          # echo "      - intelmpi" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
+
+          # For GNU
```

Comment:
We shouldn't need lines 79-94. Running `spack compiler find --scope=system` should suffice